cmd/subnetcmd/create.go

```diff
@@ -218,6 +235,48 @@ func createSubnetConfig(cmd *cobra.Command, args []string) error {
 		return errors.New("not implemented")
 	}
 
+	if teleporterReady {
+		isSubnetEVMGenesis, err := jsonIsSubnetEVMGenesis(genesisBytes)
+		if err != nil {
+			return err
+		}
+		if !isSubnetEVMGenesis {
+			return fmt.Errorf("teleporter is not supported yet for non Subnet-EVM virtual machines")
+		}
+		keyPath := app.GetKeyPath(constants.TeleporterKeyName)
+		var k *key.SoftKey
+		if utils.FileExists(keyPath) {
+			ux.Logger.PrintToUser("loading stored key %q for teleporter deploys", constants.TeleporterKeyName)
+			k, err = key.LoadSoft(models.LocalNetwork.ID, keyPath)
+			if err != nil {
+				return err
+			}
+		} else {
+			ux.Logger.PrintToUser("generating stored key %q for teleporter deploys", constants.TeleporterKeyName)
+			k, err = key.NewSoft(0)
+			if err != nil {
+				return err
+			}
+			if err := k.Save(keyPath); err != nil {
+				return err
+			}
+		}
+		ux.Logger.PrintToUser("  (evm address, genesis balance) = (%s, %v)", k.C(), teleporter.TeleporterPrefundedAddressBalance)
+		genesisBytes, err = addSubnetEVMGenesisPrefundedAddress(genesisBytes, k.C(), teleporter.TeleporterPrefundedAddressBalance.String())
+		if err != nil {
+			return err
+		}
+		// lets' use latest versions for teleporter contract and default awm relayer
+		teleporterVersion, err := app.Downloader.GetLatestReleaseVersion(binutils.GetGithubLatestReleaseURL(constants.AvaLabsOrg, constants.TeleporterRepoName))
+		if err != nil {
+			return err
+		}
```

Comment:
There is no compatiblity file. I believe may not be needed. It is mainly evm code and related to the underlying warp precompile. Surely there is a dependency between warp implementation and teleporter implementation. Not sure if warp implementation is completelly fixed? cc @michaelkaplan13

Reply:
Yep, that is correct. Teleporter only depends on the warp precompile being available.

I'm hesitant to say anything is completely fixed, but the warp precompile can only be changed with a required network upgrade, and any non-backwards compatible changes would potentially break any contracts that use it (including Teleporter), so that would be avoid at the EVM level at basically all costs.

TL;DR: I don't think you need to worry about version compatibility here as long as the blockchain has the warp precompile enabled. 👍